File: code_samples/typescript/oop_interfaces.ts
/**
 * Oop: Interfaces
 * AI/ML Training Sample
 */

interface IInterfaces {
    data: string;
    process(input: string): void;
    validate(): boolean;
}

class Interfaces implements IInterfaces {
    data: string;
    
    constructor() {
        this.data = "";
    }
    
    process(input: string): void {
        this.data = input;
    }
    
    getData(): string {
        return this.data;
    }
    
    validate(): boolean {
        return this.data.length > 0;
    }
}

// Example usage
const instance = new Interfaces();
instance.process("example");
console.log(`Data: ${instance.getData()}`);
console.log(`Valid: ${instance.validate()}`);

export { Interfaces, IInterfaces };

File: code_samples/elixir/oop_interfaces.ex
# Oop: Interfaces
# AI/ML Training Sample

defmodule Interfaces do
  defstruct data: ""
  
  def new(), do: %Interfaces{}
  
  def process(%Interfaces{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%Interfaces{data: data}), do: data
  
  def validate(%Interfaces{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = Interfaces.new()
updated = Interfaces.process(instance, "example")
IO.puts("Data: " <> Interfaces.get_data(updated))
IO.puts("Valid: " <> to_string(Interfaces.validate(updated)))
